docs/proposals/petset.md

```diff
@@ -0,0 +1,533 @@
+as a number:
+
+```
+// Proposal 2f: variant: specify merge-patch semantics with some variable replacement as an integer
+{
+    "spec": {
+        // ...
+
+        "supportTemplates": [{
+            "replicationController": {
+                "spec": {"replicas": 1}
+            },
+            // "$(REPLICAS)" is transformed into 5, not "5"
+            "transform": [{"op": "add", "path": "/metadata/name", "value": "$(REPLICAS)", type: "number"}]
+        }]
+    }
+}
+
+```
+
+
+
+### Accessing pods by stable network identity
+
+In order to provide stable network identity, given that pods may not assume pod IP is constant over the
+lifetime of a pod, it must be possible to have a resolvable DNS name for the pod that is tied to the
+pod identity. There are two broad classes of clustered services - those that require clients to know
+all members of the cluster (load balancer intolerant) and those that are amenable to load balancing.
+For the former, clients must also be able to easily enumerate the list of DNS names that represent the
+member identities and access them inside the cluster. Within a pod, it must be possible for containers
+to find and access that DNS name for identifying itself to the cluster.
+
+Since a pod is expected to be controlled by a single controller at a time, it is reasonable for a pod to
+have a single identity at a time. Therefore, a service can expose a pod by its identity in a unique
+fashion via DNS by leveraging information written to the endpoints by the endpoints controller. The
+mechanism for that is either generic metadata delivered to the pod by the PetSet controller (an identity
+field as in 1c, or an annotation as in 1b).
+
+The end result might be DNS resolution as follows:
+
+```
+# service mongo pointing to pods created by PetSet mdb, with identities mdb-1, mdb-2, mdb-3
+
+dig mongodb.namespace.svc.cluster.local +short A
+172.130.16.50
+
+dig mdb-1.mongodb.namespace.svc.cluster.local +short A
+# IP of pod created for mdb-1
+
+dig mdb-2.mongodb.namespace.svc.cluster.local +short A
+# IP of pod created for mdb-2
+
+dig mdb-3.mongodb.namespace.svc.cluster.local +short A
+# IP of pod created for mdb-3
+
+# made up name, returns a set of CNAMES? or SRV records for each identity ???
+dig mongodb.namespace.identity.cluster.local CNAME
+mdb-1.mongodb.namespace.svc.cluster.local
+mdb-2.mongodb.namespace.svc.cluster.local
+mdb-3.mongodb.namespace.svc.cluster.local
+```
+
+
+### Preventing duplicate identities
+
+The PetSet controller is expected to execute like other controllers, as a single writer.  However, when
+considering designing for safety first, the possibility of the controller running concurrently cannot
+be overlooked, and so it is important to ensure that duplicate pod identities are not achieved.
+
+There are two mechanisms to acheive this at the current time. One is to leverage unique names for pods
```

Comment:
I would say that if we are specifying hostname-template in the podSec, then kube2sky should automatically use that as the record name, independent of whether its a petset pod or not.